index.php

```diff
@@ -188,6 +188,34 @@ class Webgrind_MasterConfig
             }
         break;
 
+        case 'download_link':
+            $file = Webgrind_Config::exposeServerFile(get('file'));
+
+            if (empty($file)) {
+                sendJson(array('error' => 'No file found or access denied!'));
+                exit;
+            }
+
+            $params = array('op' => 'download_file', 'file' => get('file'));
+            sendJson(array('done' => "?".http_build_query($params)));
+        break;
+
+        case 'download_file':
+            $file = Webgrind_Config::exposeServerFile(get('file'));
+
+            if (empty($file)) {
+                exit;
+            }
+
+            header("Cache-Control: public");
+            header("Content-Description: File Transfer");
+            header("Content-Disposition: attachment; filename=".get('file'));
+            header("Content-Type: text/plain");
+            header("Content-Transfer-Encoding: binary");
+
+            readfile($file);
```

Comment:
Hmm, I get "No such file or directory" errors.  The following appears to fix it for me, can you confirm?
```diff
diff --git a/index.php b/index.php
index f127542..5d1825f 100644
--- a/index.php
+++ b/index.php
@@ -189,7 +189,7 @@ try {
         break;
 
         case 'download_link':
-            $file = Webgrind_Config::exposeServerFile(get('file'));
+            $file = Webgrind_Config::exposeServerFile(Webgrind_Config::xdebugOutputDir().get('file'));
 
             if (empty($file)) {
                 sendJson(array('error' => 'No file found or access denied!'));
@@ -201,7 +201,7 @@ try {
         break;
 
         case 'download_file':
-            $file = Webgrind_Config::exposeServerFile(get('file'));
+            $file = Webgrind_Config::exposeServerFile(Webgrind_Config::xdebugOutputDir().get('file'));
 
             if (empty($file)) {
                 exit;
```

Reply:
That would ultimately depend on the `exposeServerFile()` implementation in my opinion.
If it requires a full file path then that would certainly fix it, I'm not quite sure how it's supposed to be used to be honest.

Reply:
While testing this earlier I returned the full path for the file from the `exposeServerFile(...)`-function. I realise now that it was incorrect. I have made your suggested changes and some cleaning up to match the surrounding coding style.